compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -470,9 +468,11 @@ fn check_opaque_meets_bounds<'tcx>(
         hir::OpaqueTyOrigin::TyAlias { .. }
             if tcx.def_kind(tcx.parent(def_id.to_def_id())) == DefKind::OpaqueTy => {}
         // Can have different predicates to their defining use
```

Comment:
We could, perhaps, define the assumed wf types of a TAIT to be any types that show up in the bounds of a parent TAIT.

Don't think it's worth unless someone can come up with an unsoundness test here, though.